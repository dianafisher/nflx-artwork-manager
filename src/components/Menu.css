.menu {
  flex-shrink: 0;
  width: 300px;
  height: 500px;
  background-color: #fff;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.2), 0 1px 5px 0 rgba(0,0,0,.12);
}

.menu-title {
  background-color: #df2230; /*#00bcd4;*/
  height: 48px;
  font-size: 20px;
  color: #fff;
  display: flex;
  align-items: center;
  padding-left: 20px;
}

.menu-button {
  background-color: #df2230;
  color: rgba(255,255,255,.84);
  border-radius: 5px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.2), 0 1px 5px 0 rgba(0,0,0,.12);
  padding: 8px 30px;
  margin: 10px 1px;
  font-size: 14px;
  text-transform: uppercase;
  border: none;
}

.menu-subtitle {
  color: #df2230;
  font-weight: 300;
  font-size: 18px;
  text-align: left;
  padding-left: 20px;
}

div.form-group {
  padding-left: 15px;
}

.form-group label {
  color: #9e9e9e;
}

.form-horizontal .radio {
  min-height: 30px;
  padding-top: 8px;
}

.form-horizontal label {
  text-align: right;
}

.checkbox, .radio {
  position: relative;
  display: block;
}

.radio label {
  cursor: pointer;
  padding-left: 45px;
  min-height: 22px;
  font-size: 16px;
  line-height: 1.42857;
  font-weight: 400;
}

.radio span {
  display: block;
  position: absolute;
  left: 9px;
  top: 2px;
  transition-duration: .2s;
}

.radio input[type=radio] {
  position: absolute;
  left: 13px;
  top: 5px;
}

.radio input[type=radio]:checked~.check {
  background-color: #df2230;
}

.radio .check {
  height: 14px;
  width: 14px;
  border-radius: 100%;
  position: absolute;
  left: 14px;
  top: 7px;
}

.radio .circle, label.radio-inline .circle {
  border: 2px solid rgba(0,0,0,.54);
  height: 20px;
  width: 20px;
  border-radius: 100%;
}

span.text {
  margin-left: 32px;
  transition-duration: .2s;
}

span.text:hover {
  color: #616161;
}
