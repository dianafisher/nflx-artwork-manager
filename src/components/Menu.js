import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Menu.css';

class Menu extends Component{

  _onChange = (e) => {
    this.props.onGroupByChange(e.target.value);
  }

  _renderRadio = (value, text) => {
    const groupBy = this.props.groupedBy;
    return (
      <div className='radio'>
        <label>
          <input
            onChange={this._onChange}
            type="radio"
            name="groupOptions"
            value={value}
            checked={groupBy === value}
          ></input>
          <span className='circle'></span>
          <span className='check'></span>
          <span className='text'>{text}</span>
        </label>
      </div>
    )
  }

  render() {

    return (
      <div className='menu'>
        <div className='menu-title'>Options</div>
        <div className='menu-content'>
          <form className='form-horizontal'>
            <h4 className='menu-subtitle'>Group By</h4>
            <div className='form-group'>
              {this._renderRadio('movieId', 'Movie')}
              {this._renderRadio('languageCode', 'Language')}
            </div>
          </form>
        </div>
      </div>
    )
  }

}

Menu.propTypes = {
  onGroupByChange: PropTypes.func.isRequired,
  groupedBy: PropTypes.string.isRequired
};

export default Menu;
