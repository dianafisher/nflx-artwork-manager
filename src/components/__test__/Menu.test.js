import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../Menu';

describe('Menu', () => {

  it('renders', () => {
    expect(shallow(
      <Menu onGroupByChange={jest.fn()} groupedBy='movieId'></Menu>
    )).toMatchSnapshot();
  });

  it('changes radio tune when clicked', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Menu onGroupByChange={onChange} groupedBy='movieId'></Menu>
    );

    wrapper.find('input[value="languageCode"]').simulate('change', {target: {checked: true}});
    expect(onChange).toBeCalled();
  });
});
