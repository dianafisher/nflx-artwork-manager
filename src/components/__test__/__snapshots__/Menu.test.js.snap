// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="menu"
>
    <div
        className="menu-title"
    >
        Options
    </div>
    <div
        className="menu-content"
    >
        <form
            className="form-horizontal"
        >
            <h4
                className="menu-subtitle"
            >
                Group By
            </h4>
            <div
                className="form-group"
            >
                <div
                    className="radio"
                >
                    <label>
                        <input
                            checked={true}
                            name="groupOptions"
                            onChange={[Function]}
                            type="radio"
                            value="movieId"
                        />
                        <span
                            className="circle"
                        />
                        <span
                            className="check"
                        />
                        <span
                            className="text"
                        >
                            Movie
                        </span>
                    </label>
                </div>
                <div
                    className="radio"
                >
                    <label>
                        <input
                            checked={false}
                            name="groupOptions"
                            onChange={[Function]}
                            type="radio"
                            value="languageCode"
                        />
                        <span
                            className="circle"
                        />
                        <span
                            className="check"
                        />
                        <span
                            className="text"
                        >
                            Language
                        </span>
                    </label>
                </div>
            </div>
        </form>
    </div>
</div>,
  "nodes": Array [
    <div
      className="menu"
>
      <div
            className="menu-title"
      >
            Options
      </div>
      <div
            className="menu-content"
      >
            <form
                  className="form-horizontal"
            >
                  <h4
                        className="menu-subtitle"
                  >
                        Group By
                  </h4>
                  <div
                        className="form-group"
                  >
                        <div
                              className="radio"
                        >
                              <label>
                                    <input
                                          checked={true}
                                          name="groupOptions"
                                          onChange={[Function]}
                                          type="radio"
                                          value="movieId"
                                    />
                                    <span
                                          className="circle"
                                    />
                                    <span
                                          className="check"
                                    />
                                    <span
                                          className="text"
                                    >
                                          Movie
                                    </span>
                              </label>
                        </div>
                        <div
                              className="radio"
                        >
                              <label>
                                    <input
                                          checked={false}
                                          name="groupOptions"
                                          onChange={[Function]}
                                          type="radio"
                                          value="languageCode"
                                    />
                                    <span
                                          className="circle"
                                    />
                                    <span
                                          className="check"
                                    />
                                    <span
                                          className="text"
                                    >
                                          Language
                                    </span>
                              </label>
                        </div>
                  </div>
            </form>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Menu
        groupedBy="movieId"
        onGroupByChange={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Menu {
        "_onChange": [Function],
        "_reactInternalInstance": [Circular],
        "_renderRadio": [Function],
        "context": Object {},
        "props": Object {
          "groupedBy": "movieId",
          "onGroupByChange": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="menu"
>
          <div
                    className="menu-title"
          >
                    Options
          </div>
          <div
                    className="menu-content"
          >
                    <form
                              className="form-horizontal"
                    >
                              <h4
                                        className="menu-subtitle"
                              >
                                        Group By
                              </h4>
                              <div
                                        className="form-group"
                              >
                                        <div
                                                  className="radio"
                                        >
                                                  <label>
                                                            <input
                                                                      checked={true}
                                                                      name="groupOptions"
                                                                      onChange={[Function]}
                                                                      type="radio"
                                                                      value="movieId"
                                                            />
                                                            <span
                                                                      className="circle"
                                                            />
                                                            <span
                                                                      className="check"
                                                            />
                                                            <span
                                                                      className="text"
                                                            >
                                                                      Movie
                                                            </span>
                                                  </label>
                                        </div>
                                        <div
                                                  className="radio"
                                        >
                                                  <label>
                                                            <input
                                                                      checked={false}
                                                                      name="groupOptions"
                                                                      onChange={[Function]}
                                                                      type="radio"
                                                                      value="languageCode"
                                                            />
                                                            <span
                                                                      className="circle"
                                                            />
                                                            <span
                                                                      className="check"
                                                            />
                                                            <span
                                                                      className="text"
                                                            >
                                                                      Language
                                                            </span>
                                                  </label>
                                        </div>
                              </div>
                    </form>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="menu"
>
          <div
                    className="menu-title"
          >
                    Options
          </div>
          <div
                    className="menu-content"
          >
                    <form
                              className="form-horizontal"
                    >
                              <h4
                                        className="menu-subtitle"
                              >
                                        Group By
                              </h4>
                              <div
                                        className="form-group"
                              >
                                        <div
                                                  className="radio"
                                        >
                                                  <label>
                                                            <input
                                                                      checked={true}
                                                                      name="groupOptions"
                                                                      onChange={[Function]}
                                                                      type="radio"
                                                                      value="movieId"
                                                            />
                                                            <span
                                                                      className="circle"
                                                            />
                                                            <span
                                                                      className="check"
                                                            />
                                                            <span
                                                                      className="text"
                                                            >
                                                                      Movie
                                                            </span>
                                                  </label>
                                        </div>
                                        <div
                                                  className="radio"
                                        >
                                                  <label>
                                                            <input
                                                                      checked={false}
                                                                      name="groupOptions"
                                                                      onChange={[Function]}
                                                                      type="radio"
                                                                      value="languageCode"
                                                            />
                                                            <span
                                                                      className="circle"
                                                            />
                                                            <span
                                                                      className="check"
                                                            />
                                                            <span
                                                                      className="text"
                                                            >
                                                                      Language
                                                            </span>
                                                  </label>
                                        </div>
                              </div>
                    </form>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Menu
    groupedBy="movieId"
    onGroupByChange={[Function]}
/>,
}
`;
